clarity/src/vm/ast/definition_sorter/mod.rs

```diff
@@ -334,8 +349,8 @@ impl<'a> DefinitionSorter {
             // 1. (define-public func_name body)
             // 2. (define-public (func_name (arg uint) ...) body)
             // The goal here is to traverse case 2, looking for trait references
-            if let Some((_, pairs)) = func_sig.split_first() {
-                let pairs_vec: Vec<&PreSymbolicExpression> = pairs.iter().collect();
+            if let Some((_, pairs)) = self.filter_comments(func_sig.iter()).split_first() {
+                let pairs_vec: Vec<&PreSymbolicExpression> = pairs.to_vec();
                 self.probe_for_dependencies_in_list_of_wrapped_key_value_pairs(
                     pairs_vec, tle_index, version,
                 )?;
```

Comment:
```suggestion
            if let Some((_, pairs)) = self.filter_comments(func_sig.iter()).split_first() {
                self.probe_for_dependencies_in_list_of_wrapped_key_value_pairs(
                    &pairs, tle_index, version,
                )?;
```

Reply:
Included in 718ff4077bfb9a537c7b0054b66e1dc69271c554.